crates/forge/src/lib.rs

```diff
@@ -462,35 +487,101 @@ fn run_with_fuzzing(
+        let case_copy = case.clone();
+        let runner = runner.clone();
+        let args = args.clone();
+        let cancellation_fuzzing_token = token.clone();
+        let runner_config = runner_config.clone();
+        let send = send.clone();
+
+        tasks.push(task::spawn(async move {
+            tokio::select! {
+                () = runner_config.error_cancellation_token.cancelled() => {
+                    //Stop executing all tests because flag --exit-first'
+                    //has been set and one test FAIL
+                    Ok(TestCaseSummary::None {  })
+                },
+                () = runner_config.cancellation_token.cancelled() => {
+                    //Stop executing all tests because flag --exit-first'
+                    //has been set and one test FAIL
+                    Ok(TestCaseSummary::skipped(&case_copy))
+                },
+                () = cancellation_fuzzing_token.cancelled() => {
+                    //Stop executing all tests because flag --exit-first'
+                    //has been set and one test FAIL
+                    Ok(TestCaseSummary::skipped(&case_copy))
+                },
+               result = blocking_run_from_test(
+                    runner,
+                    case,
+                    runner_config.clone(),
+                    args.clone(),
+                    Some(send),
+                ) => {
+                    match result {
+                        Ok(result) => {
+                            if let TestCaseSummary::Failed { .. } = &result {
+                                if runner_config.exit_first {
+                                    runner_config.cancellation_token.cancel();
+                                } else {
+                                    cancellation_fuzzing_token.cancel();
+                                }
+                            }
+                            Ok(result)
+                        }
+                        Err(e) => {
+                            runner_config.error_cancellation_token.cancel();
+                            Err(e)
+                        }
+                    }
+                },
+            }
+        }));
+    }
 
-        let result = run_from_test_case(
-            package_root,
-            runner,
-            case,
-            runner_config.fork_targets.as_ref(),
-            &runner_params.contracts,
-            &runner_params.predeployed_contracts,
-            args.clone(),
-            &runner_params.environment_variables,
-        )?;
+    let mut results = vec![];
+
+    drop(send);
+
+    while let Some(task) = tasks.next().await {
+        let result = task??;
         results.push(result.clone());
 
-        if let TestCaseSummary::Failed { .. } = result {
-            // Fuzz failed
-            break;
+        match &result {
+            TestCaseSummary::Failed { .. } | TestCaseSummary::None {} => {
+                break;
+            }
+            _ => (),
```

Comment:
I think we may not yield some tests that were executed bcs we break, but imo it's fine

Reply:
in the context of fuzzer test has to pass all "subtests", otherwise is Failed of skiped